subprojects/dependency-management/src/main/java/org/gradle/internal/resource/cached/DefaultCachedExternalResourceIndex.java

```diff
@@ -107,5 +132,23 @@ public void write(Encoder encoder, CachedExternalResource value) throws Exceptio
                 }
             }
         }
+
+        private String relativizeAndNormalizeFilePath(File cachedFile) {
```

Comment:
:o: I wonder what the performance tests will say here. Normalization and denormalization tends to appear as hotspot in flame graphs. Here we make it a bit more complex and this operation can be called quite often. I wonder if it would help splitting this method into smaller chunks so that they can be inlined.

Reply:
No complaints so far in the different runs, though it was my fear initially.